code/datums/mutations/touch.dm

```diff
@@ -85,3 +85,196 @@
 	icon = 'icons/obj/weapons/hand.dmi'
 	icon_state = "zapper"
 	inhand_icon_state = "zapper"
+
+/datum/mutation/human/lay_on_hands
+	name = "Mending Touch"
+	desc = "The affected can lay their hands on other people to transfer a small amount of their injuries to themselves."
+	quality = POSITIVE
+	locked = FALSE
+	difficulty = 16
+	text_gain_indication = "<span class='notice'>Your hand feels blessed!</span>"
+	text_lose_indication = "<span class='notice'>Your hand feels secular once more.</span>"
+	power_path = /datum/action/cooldown/spell/touch/lay_on_hands
+	instability = 35
+	energy_coeff = 1
+	power_coeff = 1
+	synchronizer_coeff = 1
+
+/datum/mutation/human/lay_on_hands/modify()
+	. = ..()
+	var/datum/action/cooldown/spell/touch/lay_on_hands/to_modify =.
+
+	if(!istype(to_modify)) // null or invalid
+		return
+
+	// More healing if powered up.
+	to_modify.heal_multiplier = GET_MUTATION_POWER(src)
+	// Less pain if synchronized.
+	to_modify.pain_multiplier = GET_MUTATION_SYNCHRONIZER(src)
+
+/datum/action/cooldown/spell/touch/lay_on_hands
+	name = "Mending Touch"
+	desc = "You can now lay your hands on other people to transfer a small amount of their physical injuries to yourself."
+	button_icon = 'icons/mob/actions/actions_genetic.dmi'
+	button_icon_state = "mending_touch"
+	sound = 'sound/magic/staff_healing.ogg'
+	cooldown_time = 12 SECONDS
+	school = SCHOOL_RESTORATION
+	invocation_type = INVOCATION_NONE
+	spell_requirements = NONE
+	antimagic_flags = NONE
+
+	hand_path = /obj/item/melee/touch_attack/lay_on_hands
+	draw_message = span_notice("You ready your hand to transfer injuries to yourself.")
+	drop_message = span_notice("You lower your hand.")
+	/// Multiplies the amount healed, without increasing the recieved damage.
+	var/heal_multiplier = 1
+	/// Multiplies the incoming pain from healing.
+	var/pain_multiplier = 1
+	/// Icon used for beaming effect
+	var/beam_icon = "blood"
+
+/datum/action/cooldown/spell/touch/lay_on_hands/cast_on_hand_hit(obj/item/melee/touch_attack/hand, atom/victim, mob/living/carbon/mendicant)
+
+	if(!iscarbon(victim))
+		return
+
+	var/mob/living/hurtguy = victim
+
+
+	// Heal more, hurt a bit more.
+	// If you crunch the numbers it sounds crazy good,
+	// but I think that's a fair reward for combining the efforts of Genetics, Medbay, and Mining to reach a hidden mechanic.
+	if(HAS_TRAIT_FROM(mendicant, TRAIT_PACIFISM, HIPPOCRATIC_OATH_TRAIT))
```

Comment:
Hippocratic oath should really apply its own trait if you need to do this.

Reply:
i dont see the reason to tbh this works

Reply:
I dont get the ask here

Reply:
watermelon says i should just add trait_aesculapius_staff

i dont see the point as we arleady have a eprfectly functinoal has trait from

Reply:
Is there any instance in which we should be using ``HAS_TRAIT_FROM`` if this isn't where it should be used?

Reply:
the reason I don't like this and general usage of HAS_TRAIT_FROM is because you are tying this behaviour to the pacifism trait when it does not deserve to be tied to the pacifism trait. Although unlikely, the pacifism from hippocratic oath could be removed for any reason. For example, someone could choose to do a rework where they allow players to take the hippocratic oath and then remove it if they commit any violent acts, instead of giving them the pacifism trait. If your goal is to determine whether they've taken the hippocratic oath, checking whether they have pacifism from the hippocratic oath is not the way to do this.

Reply:
Ultimately I don't think traits were created to be used in this way and if you need to do something like `HAS_TRAIT_FROM`, then you may just be better off splitting it into its own trait.
There are valid usecases for HAS_TRAIT_FROM, but these usecases stem from the fact that the source can dynamically change and is not static throughout the entire lifetime of the server.
I.e. `HAS_TRAIT_FROM(living, TRAIT_ON_ELEVATED_SURFACE, REF(src))`

Reply:
That makes a bit more sense, yeah.